File: Unit Area of largest region of 1's.cpp
class Solution
{
    int delrow[8]={-1,0,1,0,-1,1,1,-1};
    int delcol[8]={0,1,0,-1,-1,1,-1,1};
    public:
    //Function to find unit area of the largest region of 1s.
    int bfs(int i, int j, vector<vector<int>>&grid, vector<vector<int>>&vis) {
        
        int m = grid.size();
        int n = grid[0].size();
        
        queue<pair<int, int>>q;
        q.push({i, j});
        vis[i][j] = 1;
        int count = 0;
        while(!q.empty()) {
            int row = q.front().first;
            int col = q.front().second;
            q.pop();
            
            count++;
            
            for(int i= 0;i<8;i++) {
                int nrow = row + delrow[i];
                int ncol = col + delcol[i];
                
                if(nrow >= 0 and nrow < m and ncol >= 0 and ncol < n and vis[nrow][ncol] == 0 and grid[nrow][ncol]==1) {
                    q.push({nrow, ncol});
                    vis[nrow][ncol] = 1;
                }
            }
        }
        return count;
    }
    int findMaxArea(vector<vector<int>>& grid) {
        // Code here
        int m = grid.size();
        int n = grid[0].size();
        int maxi = 0;
        vector<vector<int>>vis(m+1, vector<int>(n+1, 0));
        for(int i= 0;i<m;i++) {
            for(int j= 0;j<n;j++) {
                if(grid[i][j]== 1 and vis[i][j] == 0) {
                    maxi = max(maxi, bfs(i, j, grid, vis));
                }
            }
        }
        return maxi;
    }
};

File: Unit Area of largest region of 1's.java
class Solution
{
    static int count = 0;
    public int findMaxArea(int[][] grid)
    {
        
        int max = 0;
       int N = grid.length, M = grid[0].length;
        int[][] visited = new int[N][M];
        for(int i = 0;i<grid.length;i++){
            for(int j = 0;j<grid[0].length;j++){
                if(grid[i][j] == 1){
                    dfs(grid,i,j,visited);
                    max = Math.max(max,count);
                    count = 0;
                }
            }
        }
        
        return max;
    }
    
    public static void dfs(int[][] grid,int i,int j,int[][] visited){
         visited[i][j] = 1;
        count++;
        int  N = grid.length, M = grid[0].length;
        int [] delRow = {-1, -1, -1, 0, 1, 1, 1, 0 };
        int [] delCol = {-1, 0, 1, 1, 1, 0, -1, -1 };
        int NRow =0, NCol = 0;
        for(int k=0; k<8; k++){
            NRow = delRow[k] + i;
            NCol = delCol[k] + j;
            if(NRow >= 0 && NRow < N && NCol >=0 && NCol < M 
            && visited[NRow][NCol] == 0 && grid[NRow][NCol] == 1){
                dfs(grid, i+delRow[k], j + delCol[k],visited);
            }
        }
    }
}
